src/databricks/labs/ucx/migration/azure_sp.py

```diff
@@ -0,0 +1,162 @@
+import csv
+import io
+import logging
+from dataclasses import dataclass
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.workspace import ExportFormat
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ServicePrincipalMigrationInfo:
+    prefix: str
+    client_id: str
+    principal: str
+    privilege: str
+    # if create access manager and managed identity for this SP
+    replace_with_ac: bool
+    # if a storage credential using this SP already exists
+    already_in_sc: bool
+    # SP's client_secret stored in Databricks secret
+    client_secret: str
+    # if this is a managed identity
+    if_mi: bool
```

Comment:
Doesn't belong to this class

Reply:
Do you want it to be moved to `assessment/azure.py`? I think this dataclass will be only used in this migration code.